lib/mixlib/install/backend/artifactory.rb

```diff
@@ -19,72 +19,141 @@
 require "net/http"
 require "json"
 require "mixlib/install/artifact_info"
+require "mixlib/install/backend/base"
+require "mixlib/versioning"
 require "artifactory"
 
 module Mixlib
   class Install
     class Backend
-      class Artifactory
-        ARTIFACTORY_ENDPOINT = "http://artifactory.chef.co".freeze
+      class Artifactory < Base
+        #
+        # Returns Array<ArtifactInfo> when platform info is not configured
+        # Returns ArtifactInfo when platform info is configured
+        #
+        def info
+          if options.platform
```

Comment:
Definitely needs some readability and improvements. Looks like we are calling `products` as well as `product_version` regardless of the case. With the current separation of logic into functions it is very hard to follow.